src/theory/theory_engine.cpp

```diff
@@ -722,7 +722,18 @@ bool TheoryEngine::presolve() {
   return false;
 }/* TheoryEngine::presolve() */
 
-void TheoryEngine::postsolve() {
+void TheoryEngine::postsolve(prop::SatValue result)
+{
+  // Handle emitting pending partitions.
+  // This can be triggered by a scatter strategy that produces
+  // fewer than the requested number of partitions before solving
+  // the remainder of the problem.
+  if (options().parallel.computePartitions > 1
```

Comment:
Let's also make this so TheoryEngineModules have a postsolve, where emitRemainingPartitions is called via PartitionGenerator::postsolve.

Reply:
Done, thanks for the speedy response!